src/Services/PrescriptionService.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/**
+ * PrescriptionService
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Yash Bothra <yashrajbothra786gmail.com>
+ * @copyright Copyright (c) 2020 Yash Bothra <yashrajbothra786gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Services;
+
+use OpenEMR\Common\Uuid\UuidRegistry;
+use OpenEMR\Validators\ProcessingResult;
+use OpenEMR\Validators\BaseValidator;
+
+class PrescriptionService extends BaseService
+{
+
+    private const PRESCRIPTION_TABLE = "prescriptions";
+    private $uuidRegistry;
+
+    /**
+     * Default constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct(self::PRESCRIPTION_TABLE);
+        $this->uuidRegistry = new UuidRegistry(['table_name' => self::PRESCRIPTION_TABLE]);
+        $this->uuidRegistry->createMissingUuids();
+    }
+
+    /**
+     * Returns a list of prescription matching optional search criteria.
+     * Search criteria is conveyed by array where key = field/column name, value = field value.
+     * If no search criteria is provided, all records are returned.
+     *
+     * @param  $search search array parameters
+     * @param  $isAndCondition specifies if AND condition is used for multiple criteria. Defaults to true.
+     * @return ProcessingResult which contains validation messages, internal error messages, and the data
+     * payload.
+     */
+    public function getAll($search = array(), $isAndCondition = true)
+    {
+        $sqlBindArray = array();
+
+        $sql = "SELECT prescriptions.*,
+                patient.uuid AS puuid,
+                encounter.uuid AS euuid,
+                practitioner.uuid AS pruuid,
+                drug.uuid AS drug_uuid
+                FROM prescriptions
+                LEFT JOIN patient_data AS patient
+                ON patient.pid = prescriptions.patient_id
+                LEFT JOIN form_encounter AS encounter
+                ON encounter.encounter = prescriptions.encounter
+                LEFT JOIN users AS practitioner
+                ON practitioner.id = prescriptions.provider_id
+                LEFT JOIN drugs AS drug
+                ON drug.drug_id = prescriptions.drug_id";
```

Comment:
Pretty sure all the potential uuid's in here need to be populated in __contruct , correct?